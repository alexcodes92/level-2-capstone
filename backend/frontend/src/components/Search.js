import React, { useState } from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";

const Search = ({ onSearch }) => {
  const [term, setTerm] = useState("");
  const [media, setMedia] = useState("all");

  // when the search form is submitted with a search term
  // call the onSearch function passed in from App.js
  const submitSearch = (e) => {
    e.preventDefault();
    if (term.length) {
      onSearch(term, media);
    }
  };

  return (
    <Form id="formName" onSubmit={submitSearch}>
      <Container>
        <Row>
          <Col xs={7}>
            <Form.Group>
              <Form.Control
                type="text"
                placeholder="Search term..."
                id="term"
                name="term"
                onChange={(e) => setTerm(e.target.value)}
                className="mb-2"
                value={term}
              />
            </Form.Group>
          </Col>
          <Col xs={3}>
            <Form.Group>
              <Form.Control
                as="select"
                onChange={(e) => setMedia(e.target.value)}
              >
                <option value="all">Search all type...</option>
                <option value="movie">Movie</option>
                <option value="podcast">Podcast</option>
                <option value="music">Music</option>
                <option value="audiobook">Audiobook</option>
                <option value="shortFilm">Short Film</option>
                <option value="tvShow">TV Show</option>
                <option value="software">Software</option>
                <option value="ebook">ebook</option>
              </Form.Control>
            </Form.Group>
          </Col>
          <Col xs={2}>
            <Button className="w-100" variant="primary" type="submit">
              Search
            </Button>
          </Col>
        </Row>
      </Container>
    </Form>
  );
};

export default Search;
