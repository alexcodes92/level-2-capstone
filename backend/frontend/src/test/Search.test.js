import React from "react";
import Search from "../components/Search";
import renderer from "react-test-renderer";

test("snapshot test", () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});
