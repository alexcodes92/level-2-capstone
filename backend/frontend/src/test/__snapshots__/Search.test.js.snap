// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<form
  className=""
  id="formName"
  onSubmit={[Function]}
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-7"
      >
        <div
          className="form-group"
        >
          <input
            className="mb-2 form-control"
            id="term"
            name="term"
            onChange={[Function]}
            placeholder="Search term..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-3"
      >
        <div
          className="form-group"
        >
          <select
            className="form-control"
            onChange={[Function]}
          >
            <option
              value="all"
            >
              Search all type...
            </option>
            <option
              value="movie"
            >
              Movie
            </option>
            <option
              value="podcast"
            >
              Podcast
            </option>
            <option
              value="music"
            >
              Music
            </option>
            <option
              value="audiobook"
            >
              Audiobook
            </option>
            <option
              value="shortFilm"
            >
              Short Film
            </option>
            <option
              value="tvShow"
            >
              TV Show
            </option>
            <option
              value="software"
            >
              Software
            </option>
            <option
              value="ebook"
            >
              ebook
            </option>
          </select>
        </div>
      </div>
      <div
        className="col-2"
      >
        <button
          className="w-100 btn btn-primary"
          disabled={false}
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</form>
`;
